lib/segment/src/data_types/vectors.rs

```diff
@@ -221,6 +233,15 @@ impl<T: PrimitiveVectorElement> TypedMultiDenseVector<T> {
         self.inner_vector.chunks_exact_mut(self.dim)
     }
 
+    /// Consumes the multi vector and returns the underlying individual vectors
+    pub fn into_multi_vectors(self) -> Vec<Vec<T>> {
+        let mut chunks = vec![];
+        for chunk in self.inner_vector.into_iter().chunks(self.dim).into_iter() {
+            chunks.push(chunk.collect());
+        }
+        chunks
```

Comment:
🥢 Unnecessary nit: since it is already using `chunks`... 
```suggestion
        self.inner_vector
            .into_iter()
            .chunks(self.dim)
            .into_iter()
            .map(Iterator::collect)
            .collect()
```